File: samples/python/53.teams-messaging-extensions-action-preview/adaptive_card_helper.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from botbuilder.core import CardFactory
from botbuilder.schema import Attachment


def create_adaptive_card_editor(
    user_text: str = None,
    is_multi_select: bool = False,
    option1: str = None,
    option2: str = None,
    option3: str = None,
) -> Attachment:
    return CardFactory.adaptive_card(
        {
            "actions": [
                {
                    "data": {"submitLocation": "messagingExtensionFetchTask"},
                    "title": "Submit",
                    "type": "Action.Submit",
                }
            ],
            "body": [
                {
                    "text": "This is an Adaptive Card within a Task Module",
                    "type": "TextBlock",
                    "weight": "bolder",
                },
                {"type": "TextBlock", "text": "Enter text for Question:"},
                {
                    "id": "Question",
                    "placeholder": "Question text here",
                    "type": "Input.Text",
                    "value": user_text,
                },
                {"type": "TextBlock", "text": "Options for Question:"},
                {"type": "TextBlock", "text": "Is Multi-Select:"},
                {
                    "choices": [
                        {"title": "True", "value": "true"},
                        {"title": "False", "value": "false"},
                    ],
                    "id": "MultiSelect",
                    "isMultiSelect": "false",
                    "style": "expanded",
                    "type": "Input.ChoiceSet",
                    "value": "true" if is_multi_select else "false",
                },
                {
                    "id": "Option1",
                    "placeholder": "Option 1 here",
                    "type": "Input.Text",
                    "value": option1,
                },
                {
                    "id": "Option2",
                    "placeholder": "Option 2 here",
                    "type": "Input.Text",
                    "value": option2,
                },
                {
                    "id": "Option3",
                    "placeholder": "Option 3 here",
                    "type": "Input.Text",
                    "value": option3,
                },
            ],
            "type": "AdaptiveCard",
            "version": "1.0",
        }
    )


def create_adaptive_card_preview(
    user_text: str = None,
    is_multi_select: bool = False,
    option1: str = None,
    option2: str = None,
    option3: str = None,
) -> Attachment:
    return CardFactory.adaptive_card(
        {
            "actions": [
                {
                    "type": "Action.Submit",
                    "title": "Submit",
                    "data": {"submitLocation": "messagingExtensionSubmit"},
                }
            ],
            "body": [
                {
                    "text": "Adaptive Card from Task Module",
                    "type": "TextBlock",
                    "weight": "bolder",
                },
                {"text": user_text, "type": "TextBlock", "id": "Question"},
                {
                    "id": "Answer",
                    "placeholder": "Answer here...",
                    "type": "Input.Text",
                },
                {
                    "choices": [
                        {"title": option1, "value": option1},
                        {"title": option2, "value": option2},
                        {"title": option3, "value": option3},
                    ],
                    "id": "Choices",
                    "isMultiSelect": is_multi_select,
                    "style": "expanded",
                    "type": "Input.ChoiceSet",
                },
            ],
            "type": "AdaptiveCard",
            "version": "1.0",
        }
    )

File: samples/javascript_nodejs/53.teams-messaging-extensions-action-preview/adaptiveCardHelper.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

const { CardFactory } = require('botbuilder');

class AdaptiveCardHelper {
    static toSubmitExampleData(action) {
        const activityPreview = action.botActivityPreview[0];
        const attachmentContent = activityPreview.attachments[0].content;
        const userText = attachmentContent.body[1].text;
        const choiceSet = attachmentContent.body[3];
        const attributionFlag = attachmentContent.body[4].text.split(':')[1];
        return {
            MultiSelect: choiceSet.isMultiSelect ? 'true' : 'false',
            UserAttributionSelect: attributionFlag,
            Option1: choiceSet.choices[0].title,
            Option2: choiceSet.choices[1].title,
            Option3: choiceSet.choices[2].title,
            Question: userText
        };
    }

    static createAdaptiveCardEditor(userText = null, isMultiSelect = true, option1 = null, option2 = null, option3 = null) {
        return CardFactory.adaptiveCard({
            actions: [
                {
                    data: {
                        submitLocation: 'messagingExtensionFetchTask'
                    },
                    title: 'Submit',
                    type: 'Action.Submit'
                }
            ],
            body: [
                {
                    text: 'This is an Adaptive Card within a Task Module',
                    type: 'TextBlock',
                    weight: 'bolder'
                },
                { type: 'TextBlock', text: 'Enter text for Question:' },
                {
                    id: 'Question',
                    placeholder: 'Question text here',
                    type: 'Input.Text',
                    value: userText
                },
                { type: 'TextBlock', text: 'Options for Question:' },
                { type: 'TextBlock', text: 'Is Multi-Select:' },
                {
                    choices: [{ title: 'True', value: 'true' }, { title: 'False', value: 'false' }],
                    id: 'MultiSelect',
                    isMultiSelect: false,
                    style: 'expanded',
                    type: 'Input.ChoiceSet',
                    value: isMultiSelect ? 'true' : 'false'
                },
                {
                    id: 'Option1',
                    placeholder: 'Option 1 here',
                    type: 'Input.Text',
                    value: option1
                },
                {
                    id: 'Option2',
                    placeholder: 'Option 2 here',
                    type: 'Input.Text',
                    value: option2
                },
                {
                    id: 'Option3',
                    placeholder: 'Option 3 here',
                    type: 'Input.Text',
                    value: option3
                },
                { type: 'TextBlock', text: 'Do you want to send this card on behalf of the User?' },
                {
                    choices: [{ title: 'Yes', value: 'true' }, { title: 'No', value: 'false' }],
                    id: 'UserAttributionSelect',
                    isMultiSelect: false,
                    style: 'expanded',
                    type: 'Input.ChoiceSet',
                    value: isMultiSelect ? 'true' : 'false'
                }
            ],
            type: 'AdaptiveCard',
            version: '1.0'
        });
    }

    static createAdaptiveCardAttachment(data) {
        return CardFactory.adaptiveCard({
            actions: [
                { type: 'Action.Submit', title: 'Submit', data: { submitLocation: 'messagingExtensionSubmit' } }
            ],
            body: [
                { text: 'Adaptive Card from Task Module', type: 'TextBlock', weight: 'bolder' },
                { text: `${ data.Question }`, type: 'TextBlock', id: 'Question' },
                { id: 'Answer', placeholder: 'Answer here...', type: 'Input.Text' },
                {
                    choices: [
                        { title: data.Option1, value: data.Option1 },
                        { title: data.Option2, value: data.Option2 },
                        { title: data.Option3, value: data.Option3 }
                    ],
                    id: 'Choices',
                    isMultiSelect: data.MultiSelect,
                    style: 'expanded',
                    type: 'Input.ChoiceSet'
                },
                { text: 'Sending card on behalf of user is set to:'+`${ data.UserAttributionSelect }`, type: 'TextBlock', id: 'AttributionChoice' }
            ],
            type: 'AdaptiveCard',
            version: '1.0'
        });
    }
}
exports.AdaptiveCardHelper = AdaptiveCardHelper;
